src/Illuminate/Collections/Collection.php

```diff
@@ -807,6 +807,18 @@ public function prepend($value, $key = null)
         return $this;
     }
 
+    /**
+     * Push an item onto the beginning of the collection.
+     *
+     * @param  mixed  $value
+     * @param  mixed  $key
+     * @return $this
+     */
+    public function unshift($value, $key = null)
+    {
+        return $this->prepend(...func_get_args());
```

Comment:
why func_get_args

Reply:
i am trying to make the prepend method to do the work instead of replicating the prepend code into unshift

and because prepend is depending on func_get_args, i had to pass it from unshift

Reply:
i tried this approach before,

public function unshift($value, $key = null)
    {
        return $this->prepend($value,$key);
    }
    
    but calling unshift("x") .. is going to send to prepend("x",null)
    and this is different from prepend("x") and its going to act differently
    
    but with the func_get_args approach,
    calling unshift("x") is going to call prepend("x")
    which is forcing unshift and prepend to act the same